lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+    }
+   },
+   "source": [
+    "* The posterior predictive probability distribution for the test statistic $T(d)$ generated by sampling in this way is marginalized over both parameters and (replica) datasets \n",
+    "\n",
+    "* It takes into account both the uncertainty in the data (captured by the sampling distribution) _and_ the uncertainty in the parameters (propagated from our one dataset and our prior knowledge during posterior sampling)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "* Posterior predictive model checking can be seen as the Bayesian extension of classical hypothesis testing, and is a useful test of _model adequacy_ \n",
+    "\n",
+    "* As with classical hypothesis testing, a model can be discarded (or retained) on the basis of a posterior predictive model check\n",
+    "\n",
+    "* Note that we did not have to make any approximations in order to use a standard distribution for our summary $T$: we just use the posterior PDF we already have."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Test statistics $T(d,\\theta)$ that are functions of both the data and the parameters are called **discrepancy measures**.\n",
+    "\n",
+    "The maximum log-likelihood is a common example."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Discrepancy measure: $T = \\hat{\\chi}^2$\n",
+    "\n",
+    "${\\rm Pr}(T(d^{\\rm rep},\\theta)>T(d,\\theta)\\,|\\,d) \\approx 0.0$\n",
+    "\n",
+    "<img src=\"graphics/modelcheck-linear-discrepancy.png\" width=50%>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Any way we look at it, it's unlikely that we'd conclude the linear model explains these data adequately. How do we choose an alternative?\n",
+    "\n",
+    "One way to compare the fitness of models is to look at question (2) in model evaluation: **How accurately do they predict new data?**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Generalized Predictive Accuracy\n",
+    "\n",
+    "* We typically want a fit that works well with any *potential* data set, rather than just reproducing the one we have.\n",
+    "* In general, this means an \"Occam's Razor\"-like penalty for complexity should be involved (to avoid focusing on models that \"over-fit\" the data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "In our example, we might add a quadratic term to the model: $y = b + m x + q x^2$. How do we quantify the improvement?\n",
```

Comment:
See?!?! This could have simply been `a+bx+cx^2`!

Reply:
Yes yes